tests/test_job_manager.py

```diff
@@ -8,47 +8,83 @@
-                            Session=DEFAULT,
-                            calculate_file_access_time=DEFAULT) as db_mock:
-            job_id = job_manager.execute()
-            kubernetes_client.create_namespaced_job.assert_called_once()
-            body = kubernetes_client.create_namespaced_job.call_args[1]['body']
-            env_vars = body['spec']['template']['spec']['containers'][0]['env']
-            image = body['spec']['template']['spec']['containers'][0]['image']
-            command = \
-                body['spec']['template']['spec']['containers'][0]['command']
-            assert len(env_vars) == 1
-            assert env_vars[0]['name'] == expected_env_var_name
-            assert env_vars[0]['value'] == expected_env_var_value
-            assert image == expected_image
-            assert command == expected_command
+        kubernetes_job_id = job_manager.execute()
+        created_job = Session.query(Job).filter_by(
+            backend_job_id=kubernetes_job_id).one_or_none()
+        assert created_job
+        assert created_job.docker_img == expected_image
+        assert created_job.cmd == json.dumps(expected_command)
+        assert created_job.env_vars == json.dumps(
+            {expected_env_var_name: expected_env_var_value})
+        kubernetes_client.create_namespaced_job.assert_called_once()
+        body = kubernetes_client.create_namespaced_job.call_args[1]['body']
+        env_vars = body['spec']['template']['spec']['containers'][0]['env']
+        image = body['spec']['template']['spec']['containers'][0]['image']
+        command = \
+            body['spec']['template']['spec']['containers'][0]['command']
+        assert len(env_vars) == 1
+        assert env_vars[0]['name'] == expected_env_var_name
+        assert env_vars[0]['value'] == expected_env_var_value
+        assert image == expected_image
+        assert command == expected_command
 
 
-def test_stop_kubernetes_job():
+def test_stop_kubernetes_job(app, session, sample_serial_workflow_in_db,
+                             sample_workflow_workspace):
     """Test stop of Kubernetes job."""
-    pass
+    workflow_uuid = sample_serial_workflow_in_db.id_
+    next(sample_workflow_workspace(
+        str(workflow_uuid)))
+    expected_env_var_name = "env_var"
+    expected_env_var_value = "value"
+    expected_image = "busybox"
+    expected_command = ["ls"]
+    job_manager = KubernetesJobManager(
+        docker_img=expected_image, cmd=expected_command,
+        env_vars={expected_env_var_name: expected_env_var_value},
+        workflow_uuid=workflow_uuid)
+    with mock.patch("reana_job_controller.kubernetes_job_manager."
+                    "current_k8s_batchv1_api_client") as kubernetes_client:
+        kubernetes_job_id = job_manager.execute()
+        created_job = Session.query(Job).filter_by(
+            backend_job_id=kubernetes_job_id).one_or_none()
+        assert created_job
```

Comment:
Here we should test that the job was marked as deleted. It is going to fail because we are not doing it 🤒 .

Reply:
job status is being changed by [`` k8s_watch_jobs()``](https://github.com/reanahub/reana-job-controller/blob/f503065a334a4b5a92cb72828c4771da624da2e2/reana_job_controller/k8s.py#L105), so I can't check it now.

Reply:
👍 Lets then open an issue about this `JOB_DB` versus database issues once we merge :)